Jenkinsfile

```diff
@@ -1,27 +1,53 @@
 library 'LEAD'
 pipeline {
-    agent {
+  agent any
+  environment {
+    VERSION = version()
+  }
+  stages {
+    stage('Validate Terraform') {
+      agent {
         label "lead-toolchain-terraform"
+      }
+      steps {
+        notifyPipelineStart()
+        notifyStageStart()                                                                                                                                                                      
+        container('terraform') {
+          sh "make validate"
+        }
+        notifyStageEnd([status: "Validated terraform for product version: ${VERSION}"])
+      }
+      post {
+        failure {
+          notifyStageEnd([result: "fail"])
+        }
+      }
     }
-    stages {
-        stage('Validate Terraform') {
-            steps {
-                notifyPipelineStart()
-                notifyStageStart()                                                                                                                                                                      
-                container('terraform') {
-                    script {
-                          sh "make validate"
-                    }
-                }
-            }
-            post {
-              success {
-                notifyStageEnd()
-              }
-            failure {
-                notifyStageEnd([result: "fail"])
-              }
-            }
+    stage('Gitops') {
+      agent {
+        label "lead-toolchain-gitops"
+      }
+      environment {
+        GITOPS_GIT_URL = "https://github.com/liatrio/lead-environments.git"
+        GITOPS_REPO_FILE = "aws/liatrio-sandbox/terragrunt.hcl"
+        GITOPS_VALUES = "inputs.product_version=${VERSION}"
+      }
+      steps {
+        notifyStageStart()
+        container('gitops') {
+          sh "/go/bin/gitops"
         }
+        notifyStageEnd([status: "Updated the product version in sandbox to: ${VERSION}"])
+      }
+      post {
+        failure {
+          notifyStageEnd([result: "fail"])
+        }
+      }
     }
+  }
+}
+def version() {
+  return sh(script: 'git describe --tags --dirty', returnStdout: true).trim();
```

Comment:
i don't think this will work because this isn't an actual tag...won't be able to be referenced from lead-environments

Reply:
I don't think I understand, Where is the product version defined then?

Reply:
Take a look at the `git describe` help especially what the --dirty argument does. I think the problem is that that git command creates a version kind of like the most recent tag but what we need in this case is the exact tag.

Reply:
Why is this a problem only for lead terraform and not the other repositories we added the gitops stage to?

Reply:
The other versions used in lead environment are used to create a docker image or helm chart and then referenced using that version number. It doesn't actually matter what the version is as long as it is unique and using a semver makes it easier to track. 

The product version is used by terragrunt to checkout the source repo to run terraform apply against so it has to be an actual branch, commit or tag to clone the repo.

Reply:
I think I fixed it, thanks for the great explanation 🤜 🤛